src/config.js

```diff
@@ -1,46 +1,52 @@
-import { fileURLToPath } from 'url'
-import path from 'path'
-import { promises as fs } from 'node:fs'
 import rc from 'rc'
+import { readAndParseJsonFile } from './utils/utils.js'
 
-function getLogLevel () {
+function getLogLevel() {
   return process.env.LOG_LEVEL || 'info'
 }
 
-async function getPackageJson () {
-  const currentFilePath = fileURLToPath(import.meta.url)
-  const currentDirPath = path.dirname(currentFilePath)
-  const pathToPackageJson = path.resolve(currentDirPath, '../package.json')
-  const fileContent = await fs.readFile(pathToPackageJson)
-  return JSON.parse(fileContent)
+async function getPackageJson() {
+  return readAndParseJsonFile('./package.json')
 }
 
-function getConcurrency () {
+function getConcurrency() {
   const config = loadConfigFromFile()
   return {
-    ctpConcurrency: parseInt(process.env.CTP_CONCURRENCY || config.ctpConcurrency, 10) || 10
+    ctpConcurrency:
+      parseInt(process.env.CTP_CONCURRENCY || config.ctpConcurrency, 10) || 10,
   }
 }
 
-function getClientConfig () {
+function getClientConfig() {
   const config = loadConfigFromFile()
   return {
     projectKey: process.env.CTP_PROJECT_KEY || config.ctpProjectKey,
     clientId: process.env.CTP_CLIENT_ID || config.ctpClientId,
     clientSecret: process.env.CTP_CLIENT_SECRET || config.ctpClientSecret,
     apiUrl:
-      (process.env.CTP_API_URL || config.ctpApiUrl)
-      || 'https://api.europe-west1.gcp.commercetools.com',
+      process.env.CTP_API_URL ||
+      config.ctpApiUrl ||
+      'https://api.europe-west1.gcp.commercetools.com',
     authUrl:
-      (process.env.CTP_AUTH_URL || config.ctpAuthUrl)
-      || 'https://auth.europe-west1.gcp.commercetools.com',
+      process.env.CTP_AUTH_URL ||
+      config.ctpAuthUrl ||
+      'https://auth.europe-west1.gcp.commercetools.com',
+  }
+}
+
+function getSubscriptionSetupConfig() {
+  return {
+    existingOrderTypeKey: process.env.EXISTING_ORDER_TYPE_KEY,
```

Comment:
There might be more than one type used during the checkout process. I guess it should be a list of keys then. Same applies for the other types below.